mock/py/mockbuild/file_downloader.py

```diff
@@ -41,28 +42,31 @@ def get(cls, pkg_url_or_local_file):
             return pkg
 
         cls._initialize()
-        try:
-            url = pkg
-            log.info('Fetching remote file %s', url)
-            req = requests.get(url)
-            req.raise_for_status()
-            if req.status_code != requests.codes.ok:
-                log.error("Can't download {}".format(url))
-                return False
-
-            filename = urlsplit(req.url).path.rsplit('/', 1)[1]
-            if 'content-disposition' in req.headers:
-                _, params = cgi.parse_header(req.headers['content-disposition'])
-                if 'filename' in params and params['filename']:
-                    filename = params['filename']
-            pkg = cls.tmpdir + '/' + filename
-            with open(pkg, 'wb') as filed:
-                for chunk in req.iter_content(4096):
-                    filed.write(chunk)
-            cls.backmap[pkg] = url
-            return pkg
-        except Exception as err:  # pylint: disable=broad-except
-            log.error('Downloading error %s: %s', url, str(err))
+        try_count = 0
+        max_retry = 3
+        timeout = 10
+        while try_count < max_retry:
+            try:
+                url = pkg
+                log.info('Fetching remote file %s', url)
+                req = requests.get(url, timeout=timeout)
+                req.raise_for_status()
+                filename = urlsplit(req.url).path.rsplit('/', 1)[1]
+                if 'content-disposition' in req.headers:
+                    _, params = cgi.parse_header(req.headers['content-disposition'])
+                    if 'filename' in params and params['filename']:
+                        filename = params['filename']
+                pkg = cls.tmpdir + '/' + filename
+                with open(pkg, 'wb') as filed:
+                    for chunk in req.iter_content(4096):
+                        filed.write(chunk)
+                cls.backmap[pkg] = url
+                return pkg
+            except requests.exceptions.RequestException as err:
+                try_count += 1
+                log.error('Downloading error %s: %s (retrying...)', url, str(err))
+                time.sleep(2 ** try_count)
```

Comment:
Thank you for the update, but can you hire a library for this, e.g. `python3-backoff` (as discussed in #1122)?

Reply:
I tried using the Python 3 backoff package to solve this problem, but the Python 3 backoff package is not available and I don't know where to add it

Reply:
That is weird, `python3-backoff` is both available in EPEL 8 and Fedora.  Where do you miss it?

Reply:
https://github.com/rpm-software-management/mock/pull/1125   This is my submission using Python 3 backoff, but there will be errors during ci checking,  py/mockbuild/file_downloader.py:7: E0401[import-error]: Unable to import 'backoff'